libsolidity/formal/CHC.cpp

```diff
@@ -739,117 +737,35 @@ set<frontend::Expression const*, CHC::IdCompare> CHC::transactionAssertions(ASTN
 	return assertions;
 }
 
-vector<smtutil::SortPointer> CHC::stateSorts(ContractDefinition const& _contract)
-{
-	return applyMap(
-		SMTEncoder::stateVariablesIncludingInheritedAndPrivate(_contract),
-		[](auto _var) { return smt::smtSortAbstractFunction(*_var->type()); }
-	);
-}
-
 smtutil::SortPointer CHC::constructorSort()
 {
 	solAssert(m_currentContract, "");
-	if (auto const* constructor = m_currentContract->constructor())
-		return sort(*constructor);
-
-	return make_shared<smtutil::FunctionSort>(
-		vector<smtutil::SortPointer>{smtutil::SortProvider::uintSort} + m_stateSorts,
-		smtutil::SortProvider::boolSort
-	);
+	return smt::constructorSort(*m_currentContract);
 }
 
 smtutil::SortPointer CHC::interfaceSort()
 {
 	solAssert(m_currentContract, "");
-	return interfaceSort(*m_currentContract);
+	return smt::interfaceSort(*m_currentContract);
 }
 
 smtutil::SortPointer CHC::nondetInterfaceSort()
 {
 	solAssert(m_currentContract, "");
-	return nondetInterfaceSort(*m_currentContract);
-}
-
-smtutil::SortPointer CHC::interfaceSort(ContractDefinition const& _contract)
-{
-	return make_shared<smtutil::FunctionSort>(
-		stateSorts(_contract),
-		smtutil::SortProvider::boolSort
-	);
-}
-
-smtutil::SortPointer CHC::nondetInterfaceSort(ContractDefinition const& _contract)
-{
-	auto sorts = stateSorts(_contract);
-	return make_shared<smtutil::FunctionSort>(
-		sorts + sorts,
-		smtutil::SortProvider::boolSort
-	);
+	return smt::nondetInterfaceSort(*m_currentContract);
 }
 
-smtutil::SortPointer CHC::arity0FunctionSort() const
-{
-	return make_shared<smtutil::FunctionSort>(
-		vector<smtutil::SortPointer>(),
-		smtutil::SortProvider::boolSort
-	);
-}
-
-/// A function in the symbolic CFG requires:
-/// - Index of failed assertion. 0 means no assertion failed.
-/// - 2 sets of state variables:
-///   - State variables at the beginning of the current function, immutable
-///   - Current state variables
-///    At the beginning of the function these must equal set 1
-/// - 2 sets of input variables:
-///   - Input variables at the beginning of the current function, immutable
-///   - Current input variables
-///    At the beginning of the function these must equal set 1
-/// - 1 set of output variables
 smtutil::SortPointer CHC::sort(FunctionDefinition const& _function)
 {
-	auto smtSort = [](auto _var) { return smt::smtSortAbstractFunction(*_var->type()); };
-	auto inputSorts = applyMap(_function.parameters(), smtSort);
-	auto outputSorts = applyMap(_function.returnParameters(), smtSort);
-	return make_shared<smtutil::FunctionSort>(
-		vector<smtutil::SortPointer>{smtutil::SortProvider::uintSort} + m_stateSorts + inputSorts + m_stateSorts + inputSorts + outputSorts,
-		smtutil::SortProvider::boolSort
-	);
+	return smt::functionSort(_function, m_currentContract);
```

Comment:
Just a small note here. We are usually doing `using namespace std;` and then using everything without `std::`. Is there some decision that we are not doing it for some other namespaces. For example here?

If smt is to generic name for namespace, maybe it could be called `solidity::smt` and to same like with std. I am not requesting this change here, just wanted to know why it is done like this?

Reply:
I see it is in `solidity::frontend::smt`. Is there a reason why this namespace is not used like others?

Reply:
I kept the `smt::` to make it explicit that it comes from the Sorts file, because this file also has some `*sort()` functions that prepare calls to the outer ones.